### DISCUSSION TITLE
"SuppressesLongPressGesture" in Capacitor

### DISCUSSION THREAD

**Discussion by christianhengst** [ionic-team/capacitor#3208] [Q&A] [answered]
I've just migrated from Cordova to Capacitor! When long-pressing somewhere in my iOS app (no matter where) haptic feedback is fired after a couple of milliseconds. When doing this in Safari on iOS the text selection is triggered. My entire app has text selection disabled via `user-select: none`

Cordova had the `<preference name="SuppressesLongPressGesture" value="true" />` feature which was working great. Is there any similar in Capacitor?

**Comment by christianhengst** [comment, marked as answer]
If anyone else is looking for that, I've found a solution by adding [custom native code](https://capacitorjs.com/docs/ios/custom-code):

```
@objc(suppressLongpressGesture:)
func suppressLongpressGesture(call: CAPPluginCall) {
    // get WKWebView
    let webView = self.bridge.getWebView();
    // recognize long-press gesture to prevent haptic feedback (selection start, standard behavior in Safari)
    let recognizeLongPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(handleLongpressGesture))
    recognizeLongPressGesture.minimumPressDuration = 0.45;
    recognizeLongPressGesture.allowableMovement = 100.0;
    // add this in main thread
      DispatchQueue.main.async {
        // add gesture recognizer from above
        webView?.addGestureRecognizer(recognizeLongPressGesture)
        // resolve plugin call
        call.resolve()
      }
}
@objc func handleLongpressGesture() {
    print("Long-press gesture suppressed")
}
```

**Reply by Nikita-schetko** [reply]
Want to add some details for answer above.
It took some time for me to figure out how to make it work, because i have no experience in developing for iOs.  I’ve managed to make it work in the following way:
1. Followed the instruction here ( https://capacitorjs.com/docs/ios/custom-code#webview-accessible-native-code ) and created sample plugin in the root of app project;
2. Added to plugin code *.swift block, suggested by christianhengst 
3. Added a method ‘suppressLongpressGesture’ to plugin *.m file
4. Called my plugin within the app from root component once;


Example files for swift (don't forget and bridge file (App-Bridging-Header.h)!!!): 
[suppressLongTouchPlugin.m.txt](https://github.com/ionic-team/capacitor/files/5972064/suppressLongTouchPlugin.m.txt)
[suppressLongTouchPlugin.swift.txt](https://github.com/ionic-team/capacitor/files/5972065/suppressLongTouchPlugin.swift.txt)

In the app: 
```
 import { Plugins } from '@capacitor/core';
 const {SuppressLongpressGesture } = Plugins;
 
 SuppressLongpressGesture.suppressLongpressGesture();
```

**Reply by Nikita-schetko** [reply]
For capacitor v3 i've created separate plugin for that purpose, so it can be installed through npm: 
https://github.com/Nikita-schetko/capacitor-suppress-longpress-gesture

**Reply by derbenoo** [reply]
**Thanks** @Nikita-schetko, your plugin solved my longpress iOS issues completely! :tada:

**Reply by vojto** [reply]
This wasn't working for me, until I found this in my configuration:

```
"allowsLinkPreview": false
```

Removing this line fixed the problem, and now it works.

**Edit:** Of course now it shows link previews. 🤦‍♂️ I ended up adding custom `uiDelegate` which forwards everything to Capacitor's uiDelegate, but implements this one function:

```swift
public func webView(_ webView: WKWebView, shouldPreviewElement elementInfo: WKPreviewElementInfo) -> Bool {
  return false
}
```

**Reply by iwhcto** [reply]
Hi @vojto I don't know much about Swift, where should this "webView" code go?
Every time I re-build iOS I need to manually add it? Tks
